### PR TITLE
Fix options set on infinity model instance (#336)

### PR DESCRIPTION
**Pull request by callovarne**

### CODE REVIEW

File: addon/utils.js [hunk 1]
```diff
@@ -23,26 +24,30 @@ export let objectAssign = Object.assign || function objectAssign(target) {
...
+ * else set to default param
+ *
+ * @method paramsCheck
+ * @param {String} key - param name
+ * @param {Object} options - parameter overrides
+ * @param {Object} extendedInfinityModel - custom infinity model
+ * @return {String} parameter value
+ */
+export function paramsCheck(key, options, extendedInfinityModel) {
+  const paramDefault = get(extendedInfinityModel.proto(), key);
```
Comment (snewcomer) [inline addon/utils.js]: Would `get(extendedInfinityModel, key);` w/o the `proto` method still work?

Comment (callovarne) [reply addon/utils.js]: Members like _perPageParam_ are defined on the class...should it work without _.proto()_? This is before the instance is created.

Comment (snewcomer) [reply addon/utils.js]: I think a test will show us our answer 😄 

You may be right though.

Comment (callovarne) [reply addon/utils.js]: @snewcomer: I'd [started on a test](https://github.com/ember-infinity/ember-infinity/pull/337/files#diff-24a65c3e3a9ddbe5e79c4a171552f6ba), but wasn't sure how all the pieces fit together. How can I set up mirage to respond to a request with the `?limit=50` parameter?

File: tests/unit/services/infinity-test.js [hunk 2]
```diff
@@ -128,6 +128,18 @@ module('Unit | Service | infinity', function(hooks) {
     assert.ok(model instanceof InfinityModel, 'returns cached model again');
   });
 
+  // TODO: set up mirage so a request with `?limit=25` works
+  skip('model instance can be customized by extending InfinityModel', function(assert) {
```
Comment (snewcomer) [inline tests/unit/services/infinity-test.js]: @callovarne I think this test and PR is good to go as long as it passes.  Finally took the time to read this pr ;)

**Pull request merged**